### DISCUSSION TITLE
How to work with Graphql Mesh along woth express and graphql-htp

### DISCUSSION THREAD

**Discussion by shadikkhan** [graphql/graphql-http#119] [Q&A]
Is there any way graphql-http can work with graphql mesh?
https://the-guild.dev/graphql/mesh/docs/getting-started/deploy-mesh-gateway

**Comment by enisdenjo** [comment]
If you're talking about the server - Mesh uses graphql-yoga as the server. It is fully compliant with the GraphQL over HTTP spec, so there should be no benefit of using graphql-http instead.

However, if you're talking about the client - they are compatible, you can use graphql-http client against a Mesh gateway.

**Reply by shadikkhan** [reply]
Thanks @enisdenjo for the reply, my situation is little unique. I am trying to integrate **appdynamics** monitoring tool in my project (which is based on graphql mesh using inbuilt yoga server) but as per appdynamics in case of graphql I have to use graphql-http packages otherwise monitoring will not be logged on appdynamics.

SO I am looking for option to somehow inject this into graphql mesh app for the sake for appdynamics

**Reply by enisdenjo** [reply]
Why do you have to use graphql-http? Whats the requirement?

**Reply by shadikkhan** [reply]
As per AppDynamics docs - https://urldefense.com/v3/__https:/docs.appdynamics.com/appd/24.x/24.2/en/application-monitoring/configure-instrumentation/transaction-detection-rules/custom-match-rules/node-js-business-transaction-detection__;!!CQl3mcHX2A!HtWF77cCHzANfzbvT5fil1alrG0upBZUvjKeCceEQiOzOJy2tFhJHTbBJBa3I9L1DIFGwpMnKwE9r5fgbM8O$


<img width="1034" alt="image" src="https://github.com/graphql/graphql-http/assets/16745242/e65837ae-2ea8-42d0-b1e3-416537709836">


As per their recommendation - [Here are the instructions for setting up GraphQL in a Node.js application](https://urldefense.com/v3/__https:/docs.appdynamics.com/appd/24.x/24.2/en/application-monitoring/configure-instrumentation/transaction-detection-rules/custom-match-rules/node-js-business-transaction-detection__;!!CQl3mcHX2A!HtWF77cCHzANfzbvT5fil1alrG0upBZUvjKeCceEQiOzOJy2tFhJHTbBJBa3I9L1DIFGwpMnKwE9r5fgbM8O$)
![image](https://github.com/graphql/graphql-http/assets/16745242/7d7b4ce0-aa8c-49ed-8ee2-9a1f336542f7)

**Reply by enisdenjo** [reply]
Hmm, I dont see no code examples that are specific to graphql-http or @apollo/server there. Can you point me to them? Sorry, I dont know what AppDynamics is so I dont have the context.

**Reply by shadikkhan** [reply]
Unfortunatly there is not much code there form appdynamics, after installing their package just have to add this at the beging when server start
`
require("appdynamics").profile({
    controllerHostName: process.env.APPD_HOSTNAME,
    controllerPort: process.env.APPD_PORT,
    controllerSslEnabled: process.env.APPD_SSL_ENABLED,
    accountName: process.env.APPD_ACCOUNT_NAME,
    accountAccessKey: process.env.APPD_ACCOUNT_KEY,
    applicationName: process.env.APPD_APP_NAME,
    tierName: process.env.APPD_TIER_NAME,
    nodeName: os.hostname(),
    proxyHost: process.env.APPD_PROXYHOST,
    proxyPort: process.env.APPD_PROXYPORT,
    enableGraphQL: true
});
`
And that's all we as a developer have to do as per there documentation - https://docs.appdynamics.com/appd/onprem/23.x/23.6/en/application-monitoring/install-app-server-agents/node-js-agent/install-the-node-js-agent-in-containers

**Reply by enisdenjo** [reply]
I think then it means you can point it to any server that is GraphQL over HTTP spec compliant. In that case, it will work with Mesh.

Have you tried?
